src/pyaro_readers/nilupmfebas/EbasPmfReader.py

```diff
@@ -0,0 +1,352 @@
+
+        _file_dummy = self.read_file_basic(filename)
+        matrix = _file_dummy.meta["matrix"]
+        vars_read_in_file = []
+        # multicolumn file: ebas var names come from _file_dummy.col_names_vars
+        for var_idx in range(len(_file_dummy.var_defs)):
+            # continue if the variable is not an actual data variable (but e.g. time)
+            if not _file_dummy.var_defs[var_idx].is_var:
+                continue
+            # skip additional fields...
+            if _file_dummy.var_defs[var_idx].name in FIELDS_TO_SKIP:
+                continue
+            # continue if the statistcs is to be ignored
+            try:
+                if (
+                    _file_dummy.var_defs[var_idx].statistics
+                    in self._opts["default"].ignore_statistics
+                ):
+                    continue
+            except KeyError:
+                # sometimes there's no statistics: pass
+                pass
+
+            # adjust unit string
+            unit = _file_dummy.var_defs[var_idx]["unit"]
+            if unit in UALIASES:
+                unit = UALIASES[unit]
+            var_name = f"{matrix}#{_file_dummy.var_defs[var_idx].name}#{unit}"
+            if vars_to_read is not None:
+                if var_name not in vars_to_read:
+                    continue
+            if var_name not in self._variables:
+                self._variables[var_name] = (
+                    var_name,
+                    unit,
+                )
+
+            var_unit = unit
+            stat_name = _file_dummy.meta["station_code"]
+            if stat_name not in self._stations:
+                country = _file_dummy.meta["station_code"][0:2]
+                # the location naming is not consistent
+                # try the two we have seen so far
+                try:
+                    lat = float(_file_dummy.meta["station_latitude"])
+                    lon = float(_file_dummy.meta["station_longitude"])
+                    alt_str = _file_dummy.meta["station_altitude"]
+                except KeyError:
+                    # might not always work either
+                    try:
+                        lat = float(_file_dummy.meta["measurement_latitude"])
+                        lon = float(_file_dummy.meta["measurement_longitude"])
+                        alt_str = _file_dummy.meta["measurement_altitude"]
+                    except KeyError:
+                        logger.info(
+                            f"no lat / lon found in file {filename}. Skipping the file..."
+                        )
+                        return None
+                try:
+                    # usually there's a blank between the value and the unit
+                    alt = float(alt_str.split(" ")[0])
+                except ValueError:
+                    # but unfortunately not always
+                    # remove all non numbers
+                    alt = float(re.sub(r"[^\d.-]+", "", alt_str))
```

Comment:
The station extraction logic should be it's own function.

Reply:
done